datafusion/physical-expr/src/aggregate/mod.rs

```diff
@@ -56,7 +56,7 @@ pub(crate) mod variance;
 /// * knows how to create its accumulator
 /// * knows its accumulator's state's field
 /// * knows the expressions from whose its accumulator will receive values
-pub trait AggregateExpr: Send + Sync + Debug {
+pub trait AggregateExpr: Send + Sync + Debug + PartialEq<dyn Any> {
```

Comment:
this is an API change. I understand why we need a PartialEq against `dyn Any` but it might be somewhat confusing to others

could you add a documentation describing how to do so (perhaps pointing at the `down_cast_any_ref` utility function)